src/lib.rs

```diff
@@ -101,9 +108,10 @@ struct RemoteState {
 
 impl SyncSession {
     async fn run_sync(&self) -> Result<()> {
-        let (tx, rx) = async_channel::bounded(128);
-        let _sub = self.doc.subscribe_local_update(Box::new(move |u| {
-            tx.send_blocking(u.clone()).unwrap();
+        let (tx, mut rx) = tokio::sync::mpsc::channel(1);
+        let _sub = self.doc.subscribe_local_update(Box::new(move |_u| {
+            info!("doc updated locally, queue update message");
+            tx.try_send(()).ok();
```

Comment:
We might be able to switch to `tokio::sync::Notify` and use `Notify::notify_waiters` here instead, and use `changed.notified()` instead of `rx.recv()` below.